plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/price/block.json

```
@@ -0,0 +1,25 @@
+{
+	"name": "woocommerce/product-price",
+	"version": "1.0.0",
+	"title": "Product Price",
+	"description": "Display the price of a product.",
+	"category": "woocommerce-product-elements",
+	"attributes": {
+		"productId": { "type": "number", "default": 0 },
+		"isDescendentOfQueryLoop": { "type": "boolean", "default": false },
+		"textAlign": { "type": "string", "default": "" },
+		"isDescendentOfSingleProductTemplate": {
+			"type": "boolean",
+			"default": false
+		},
+		"isDescendentOfSingleProductBlock": {
+			"type": "boolean",
+			"default": false
+		}
+	},
+	"usesContext": [ "query", "queryId", "postId" ],
+	"keywords": [ "WooCommerce" ],
+	"textdomain": "woocommerce",
+	"apiVersion": 3,
```

I'm not sure if it matters but _technically_ we're missing some of the configuration present [in the shared config](https://github.com/woocommerce/woocommerce/blob/b62bc4851ac006a764400b2cda2ebf7fbd140076/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/shared/config.tsx#L29-L38).

```suggestion
	"apiVersion": 3,
	"supports": {
		"html": false
	},
	"ancestor": [
		'woocommerce/all-products',
		'woocommerce/single-product',
		'woocommerce/product-template',
		'core/post-template'
	],
```

---

> Realistically speaking, we should probably move away from the shared config entirely but that's a different problem.

I figured the same. I even started doing so in https://github.com/woocommerce/woocommerce/pull/50920 (it's just a random draft). But shared config is confusing and is a source of a lot of duplication (paradoxically 😅). So basically my idea is:
1. Migrate all of the Product Elements to `block.json`
2. Once we're there, remove shared config and cleanup block definitions.

> I'm not sure if it matters but technically we're missing some of the configuration present [in the shared config](https://github.com/woocommerce/woocommerce/blob/b62bc4851ac006a764400b2cda2ebf7fbd140076/plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/shared/config.tsx#L29-L38).

Re this, good point, I can already start moving the fields over to `block.json` and remove the dependency on shared config 👍

---

After second thoughts I decided not to do it. At the moment the `block.json` properties are anyway overwritten by shared config and `supports` in `index.tsx` file [here](https://github.com/woocommerce/woocommerce/pull/50905/files#diff-fd43d1e70b547324974885be9c533448cf5f7013d3c5cea1f60a7580c96243d7R16-R20).

Hence I think it'll be better to do the cleanup once all of the blocks are migrated and having less overwrites is easier to analyse and handle IMO. I'm planning to finalise the migration and do the cleanup in next cooldown.

---

Sorry @kmanijak, I wasn't trying to nerd snipe you 😅 I was thinking that it's something we should do later. The best approach is to do the migration of our blocks and _then_ remove the shared config. This makes sure everything works and then when we pull the shared config we are in a state where we know it _should_ work.

With that in mind it probably makes sense to ensure that _all_ of the shared config is duplicated in all of the migrated JSON files. That will make sure we are able to cleanly remove it later. Assuming we do that we should also leave a comment on the shared config indicating it is deprecated and why.

---

Makes sense! Added in https://github.com/woocommerce/woocommerce/pull/50905/commits/5215dfb89546e3d12501214cade2341f8c4d5202